src/NServiceBus.Core/Routing/FileBasedDynamicRouting/AsyncTimer.cs

```diff
@@ -34,6 +34,10 @@ public void Start(Func<Task> callback, TimeSpan interval, Action<Exception> erro
 
         public Task Stop()
         {
+            if (tokenSource == null)
```

Comment:
you could still run into issues where `return task` returns `null`, or am I missing something (do we need to return the task from the start operation?)?